books/RayTracingInOneWeekend.html

```diff
@@ -502,54 +502,103 @@
+This way, we can scale up or down the image by changing the image width, and it won't throw off our
+  desired aspect ratio.
+We do have to make sure that when we solve for the image height the resulting height is at least 1.
 
 In addition to setting up the pixel dimensions for the rendered image, we also need to set up a
-virtual viewport through which to pass our scene rays. For the standard pixel spacing (equally
-spaced in both the horizontal and vertical direction), the viewport's aspect ratio should be the
-same as our rendered image. We'll just pick a viewport two units in height. We'll also set the
-distance between the projection plane and the projection point to be one unit. This is referred to
-as the _focal length_, not to be confused with the _focus distance_, which we'll present later.
+  virtual _viewport_ through which to pass our scene rays.
+The viewport is a virtual rectangle in the 3D world that contains the grid of image pixel locations.
+If pixels are spaced the same distance horizontally as they are vertically, the viewport that
+  bounds them will have the same aspect ratio as the rendered image.
+The distance between two adjacent pixels is called the pixel spacing, and square pixels is the
+  standard.
+
+To start things off, we'll choose an arbitrary viewport height of 2.0, and scale the viewport width
+  to give us the desired aspect ratio.
+Here's a snippet of what this code will look like:
+
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ C++
+    auto aspect_ratio = 16.0 / 9.0;
+    int image_width = 400;
+
+    // Calculate the image height, and ensure that it's at least 1.
+    int image_height = static_cast<int>(image_width / aspect_ratio);
+    image_height = (image_height < 1) ? 1 : image_height;
+
+    // Calculate the viewport height. Since it's real valued, we don't check for values less then 1.
+    auto viewport_height = 2.0;
+    auto viewport_width = viewport_height * (static_cast<double>(image_width)/image_height);
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    [Listing [image-setup]: Rendered image setup]
+
+(If you're wondering why we don't just use `aspect_ratio` when computing `viewport_width`, it's
+because the _actual_ aspect ratio is the integer-based ratio of image width over image height, which
+will yield zero for values less than one.)
 
-I’ll put the “eye” (or camera center if you think of a camera) at $(0,0,0)$.
+Next we will define the camera center: a point in 3D space from which all scene rays will originate
+  (this is also commonly referred to as the _eye point_).
+The vector from the camera center to the viewport center will be orthogonal to the viewport.
+We'll initially set the distance between the viewport and the camera center point to be one unit.
+This distance is often referred to as the _focal length_.
+
+I’ll put the camera center at $(0,0,0)$.
```

Comment:
"For simplicity we'll start with the camera center at $(0,0,0)$"

Reply:
I was trying to continue the tone of the original, but it's definitely all over the map throughout the book now.

Reply:
Went with your suggestion.